src/bacnet/datalink/bvlc.c

```diff
@@ -2626,20 +2626,26 @@ const char *bvlc_result_code_name(uint16_t result_code)
  * @brief Encode a BBMD Address for Network Port object
  * @param apdu - the APDU buffer
  * @param apdu_size - the APDU buffer size
- * @param ip_address - IP address and port number
+ * @param bbmd_address - HostNPort FD BBMD Address
  * @return length of the APDU buffer
  */
 int bvlc_foreign_device_bbmd_host_address_encode(
-    uint8_t *apdu, uint16_t apdu_size, const BACNET_IP_ADDRESS *ip_address)
+    uint8_t *apdu,
+    uint16_t apdu_size,
+    const BACNET_HOST_N_PORT *const bbmd_address)
 {
+#if (__STDC__) && (__STDC_VERSION__ >= 199901L)
```

Comment:
Maybe we should move 'port' to the beginning of struct BACnetHostNPort in order to use the simpler initialization and keep the code clean?

Reply:
In this case I like the c11 and later approach. Perhaps we could keep the differentiation between standards and some day the c90 would become a relict....

Reply:
Although c11 declaration initializer is more useful in some cases, when we are simply setting all the values to zero or false it is a lot easier to maintain using the ` = { 0 };` initializer.  If the `0` is giving a compiler warning, we can just shift a numeric member to the beginning of the structure.

Reply:
Nope = { 0 } works for all. Yet I prefer the c99+ structs init. Besides we'd have the code amelioration at hand when the time comes to scrap c90 compatibility. I works well w/ c99, c11, c17 it may not be the prettiest thing in the world the #if but I wouldn't like to change this part.